finagle-core/src/main/scala/com/twitter/finagle/tracing/Trace.scala

```diff
@@ -48,19 +49,20 @@ object Trace {
     "com.twitter.finagle.tracing.TraceContext"
   ) {
     private val local = new ThreadLocal[Array[Byte]] {
-      override def initialValue(): Array[Byte] = new Array[Byte](32)
+      override def initialValue(): Array[Byte] = new Array[Byte](40)
     }
 
     def marshal(id: TraceId): Buf =
       Buf.ByteArray.Owned(TraceId.serialize(id))
 
     /**
      * The wire format is (big-endian):
-     *     ''spanId:8 parentId:8 traceId:8 flags:8''
+     *     ''spanId:8 parentId:8 traceId:8 flags:8 (traceIdHigh:8)''
      */
     def tryUnmarshal(body: Buf): Try[TraceId] = {
-      if (body.length != 32)
-        return Throw(new IllegalArgumentException("Expected 32 bytes"))
+      // Allows for 64-bit or 128-bit trace identifiers
+      if (body.length != 32 && body.length != 40)
```

Comment:
Do we ever feel that we're going to have even more bytes? If the protocol supports evolving by appending more fields, it would make sense to just ensure we have enough bytes but not care if there are extra at the tail that we don't understand. That said, I hope 128-bit is enough.

Reply:
IMO, this is OK for now. I tried to get some kind of "state of the union" view of where tracing is heading and came up with nothing conclusive in 10 minutes - we'd have to do some serious digging to answer this question, I think.